contrib/drivers/openstack/cinder/cinder.go

```diff
@@ -76,6 +81,36 @@ type CinderConfig struct {
 	Pool        map[string]PoolProperties `yaml:"pool,flow"`
 }
 
+//ListPoolOpts
+type ListPoolOpts struct {
+	// ID of the tenant to look up storage pools for.
+	TenantID string `q:"tenant_id"`
+	// Whether to list extended details.
+	Detail bool `q:"detail"`
+	//Volume_Type of the StoragePool
+	Volume_Type string `q:"volume_type"`
+}
+
+//Struct of Pools listed by volumeType
+type PoolArray struct {
+	Pools []StoragePool `json:"pools"`
+}
+
+type StoragePool struct {
+	Name         string       `json:"name"`
+	Capabilities Capabilities `json:"capabilities"`
+}
+
+type Capabilities struct {
+	FreeCapacityGB  float64 `json:"free_capacity_gb"`
```

Comment:
Can you explain why the type is `float64`? Because it could cause some type changes during the conversion.

Reply:
I copy the struct from "vendor/github.com/gophercloud/gophercloud/openstack/blockstorage/extensions/schedulerstats/results.go",it was defined in this file.but I can't use it directly as some type of the struct is different from the result of the Interface call。